ENH: Running *fMRIPrep* (#182)
TheAxonLab/hcph-sops: docs/preprocessing.md

> ## Executing fMRIPrep

Reviewer: ```suggestion
## Executing *fMRIPrep*
```

> In order to run multiple sessions in parallel for a single subject, a trick needs to be applied to avoid several jobs to work simultaneously on the same file thus compromising it.

Reviewer: ```suggestion
Because *fMRIPrep* creates a single anatomical reference for all sessions, we will need first to run with ``--anat-only`` so that the results get cached and *fMRIPrep* processes do not run into race conditions.

Similarly, when running *fMRIPrep* on a dataset with many sessions, the safest approach is to submit one job per session.
```

> - [ ] First, run only the anatomical workflow through a single fMRIPrep instance to fetch the needed template from templateflow and to generate the outputs of FreeSurfer. 

Reviewer: ```suggestion
- [ ] Submit the anatomical workflow:
```

> ``` bash title="Executing anatomical workflow of fMRIPrep"
> {% include 'code/fmriprep/ss-fmriprep-anatonly.sh' %}
> ```

Reviewer: ```suggestion
    ``` bash title="Executing anatomical workflow of fMRIPrep"
    {% include 'code/fmriprep/ss-fmriprep-anatonly.sh' %}
    ```
```

> - [ ] Once that process is successfull, you can run the sessions in parallel using the `--bids-filter-file` to launch one session per job and the `--fs-subjects-dir` flag to inform the process that the anatomical derivatives already exist.

Reviewer: ```suggestion
- [ ] Submit a *job array* with one scanning session each with the `--bids-filter-file` argument selecting the corresponding sessions, and point the `--fs-subjects-dir` argument to the folder where *FreeSurfer* results were stored.
```

> ``` bash title="Launch each session through fMRIPrep in parallel"
> cd code/fmriprep
> bash submit-fmriprep.sh
> ```

Reviewer: ```suggestion
    ``` bash title="Launch each session through fMRIPrep in parallel"
    cd code/fmriprep
    bash submit-fmriprep.sh
    ```
```

## Executing *fMRIPrep*

> Because *fMRIPrep* creates a single anatomical reference for all sessions, we will need first to run with ``--anat-only`` so that the results get cached and *fMRIPrep* processes do not run into race conditions.
> 
> Similarly, when running *fMRIPrep* on a dataset with many sessions, the safest approach is to submit one job per session.

Reviewer: ```suggestion
Because *fMRIPrep* creates a single anatomical reference for all sessions, we generate such reference first by setting the `--anat-only` flag.
If that *fMRIPrep* execution finishes successfully, the anatomical processing outcomes will be stored in the output folder.
We will then run one *fMRIPrep* process for each dataset's session, which is the recommended way for datasets with a large number of sessions (e.g., more than six sessions).
We avert that session-wise *fMRIPrep*'s processes run into race conditions by pre-computing the anatomical reference.
```

- [ ] Submit the anatomical workflow:

>     ``` bash title="Executing anatomical workflow of fMRIPrep"
>     {% include 'code/fmriprep/ss-fmriprep-anatonly.sh' %}
>     ```

Reviewer: You need to indent the file for it to render properly:

```suggestion
    ``` bash title="Executing anatomical workflow of fMRIPrep"
    {% filter indent(width=4) %}
    {% include 'code/fmriprep/ss-fmriprep-anatonly.sh' %}
    {% endfilter %}
    ```
```

- [ ] Submit a *job array* with one scanning session each with the `--bids-filter-file` argument selecting the corresponding sessions, and point the `--fs-subjects-dir` argument to the folder where *FreeSurfer* results were stored.
    ``` bash title="Launch each session through fMRIPrep in parallel"
    cd code/fmriprep
    bash submit-fmriprep.sh

>     ```

Reviewer: ```suggestion
    ```

    ??? important "The sbatch file to run fMRIPrep session-wise"

        ``` bash title="Executing anatomical workflow of fMRIPrep"
        {% filter indent(width=8) %}
        {% include 'code/fmriprep/ss-fmriprep.sh' %}
        {% endfilter %}
        ```

```
